### PR TITLE
Further Dockerfile tweaks for Heroku build

### PR DESCRIPTION
**Pull request by zerolab**
Follow up to #12. It turns out Heroku doesn't support HEREDOC either

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,33 +51,37 @@ RUN <<EOF
...
-    # Create the unprivileged user and group. If you have issues with file
-    # ownership, you may need to adjust the UID and GID build args to match your
-    # local user.
-    groupadd --gid $GID $USERNAME
-    useradd --gid $GID --uid $UID --create-home $USERNAME
-    python -m venv --upgrade-deps $VIRTUAL_ENV
-    chown -R $UID:$GID /app $VIRTUAL_ENV
-EOF
+# TODO: when moving to ONS infrastructure, replace the RUN command with
+# RUN <<<OEF
```
Comment (ababic) [inline Dockerfile:62]: Do we have a superfluous `<` here?

File: Dockerfile [hunk 2]
```diff
@@ -32,18 +32,17 @@
 
 FROM python:3.12-slim AS base
 
-ARG DOCKER_BUILDKIT
-
 WORKDIR /app
 
 # Install common OS-level dependencies
-# TODO: when moving to ONS infrastructure, replace RUN with
+# TODO: when moving to ONS infrastructure, replace RUN
```
Comment (ababic) [inline Dockerfile:38]: Was this intentional?

File: Dockerfile [hunk 3]
```diff
@@ -103,11 +107,11 @@ USER $USERNAME
 COPY pyproject.toml poetry.lock ./
 # TODO: when moving to ONS infrastructure, replace RUN with
 # RUN --mount=type=cache,target=/home/$USERNAME/.cache/,uid=$UID,gid=$GID \
-#     <<EOF
-RUN <<EOF
-    # Install the production dependencies
-    poetry install --no-root --without dev
-EOF
+#     <<< EOF
```
Comment (ababic) [inline Dockerfile:110]: Do we have a superfluous `<` here?

File: Dockerfile (hunk 1 above)
Comment (RealOrangeOne) [reply Dockerfile:62]:
```suggestion
# RUN <<EOF
```

File: Dockerfile (hunk 3 above)
Comment (RealOrangeOne) [reply Dockerfile:110]:
```suggestion
#     << EOF
```

File: Dockerfile (hunk 2 above)
Comment (zerolab) [reply Dockerfile:38]: it was, but on reflection I should revert it

File: Dockerfile (hunk 3 above)
Comment (zerolab) [reply Dockerfile]: Old habits die hard. 🦅 👁️

**Comment by zerolab** [comment]
@ababic @RealOrangeOne I've addressed your queries/suggestions

**Review by ababic — APPROVED** [review]

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**